src/v2/Apps/Artwork/Components/ArtworkSidebar/ArtworkSidebarCommercial.tsx

```diff
@@ -357,21 +357,48 @@ export class ArtworkSidebarCommercialContainer extends React.Component<
 
   render() {
     const { artwork, inquiryComponent } = this.props
+    const isPriceListed = !artwork?.isPriceHidden
+    const isOfferableFromInquiry = artwork?.isOfferableFromInquiry
+    const isOfferable = artwork?.is_offerable
+    const isAcquireable = artwork?.is_acquireable
+    const isInquireable = artwork?.is_inquireable
 
     const {
       isCommittingCreateOrderMutation,
       isCommittingCreateOfferOrderMutation,
       selectedEditionSet,
     } = this.state
 
-    const artworkEcommerceAvailable = !!(
-      artwork.is_acquireable || artwork.is_offerable
-    )
+    const artworkEcommerceAvailable = !!(isAcquireable || isOfferable)
 
-    if (!artwork.sale_message && !artwork.is_inquireable) {
+    if (!artwork.sale_message && !isInquireable) {
       return <Separator />
     }
 
+    const shouldDisplayMakeOfferButton = (): boolean => {
+      if (isOfferable || (isPriceListed && isOfferableFromInquiry)) {
+        return true
+      }
+      return false
+    }
+
+    const shouldDisplayMakeOfferAsPrimary = (): boolean => {
+      if (
+        shouldDisplayMakeOfferButton() &&
+        shouldDisplayContactGalleryButton()
+      ) {
+        return true
+      }
+      return false
+    }
```

Comment:
Similarly:
`const shouldDisplayMakeOfferAsPrimary = () => shouldDisplayMakeOfferButton() && shouldDisplayContactGalleryButton()`
...I think 🤔

Reply:
Same here, trying to avoid TS errors with an explicit return if the condition isn't met. https://ibb.co/k13CPKt